crates/ruff_linter/resources/test/fixtures/ruff/RUF011.py

```diff
@@ -1,13 +1,17 @@
 data = ["some", "Data"]
+constant = 5
 
-# Ok
+# OK
 {value: value.upper() for value in data}
 {value.lower(): value.upper() for value in data}
-{v: v*v for v in range(10)}
-{(0, "a", v): v*v for v in range(10)} # Tuple with variable
+{v: v * v for v in range(10)}
+{(0, "a", v): v * v for v in range(10)}  # Tuple with variable
+{constant: value.upper() for value in data for constant in data}
 
 # Errors
 {"key": value.upper() for value in data}
 {True: value.upper() for value in data}
 {0: value.upper() for value in data}
-{(1, "a"): value.upper() for value in data} # constant tuple
+{(1, "a"): value.upper() for value in data}  # Constant tuple
+{constant: value.upper() for value in data}
+{constant + constant: value.upper() for value in data}
```

Comment:
awesome :tada:, so quick :)

I dont know if this is a common use case but I guess one case not covered here is if the key in an attr.
e.g.
```python
{SomeClass.CONSTANT: val for val in data}
{Some.Class.CONSTANT: val for val in data}
```

Reply:
Good call, let me test that!

Reply:
See: https://github.com/astral-sh/ruff/pull/9416

Reply:
Awesome